interview-questions.md

```diff
@@ -8,47 +8,63 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 
 
 4. What is the difference between a function and a method?
 
-  Your answer:
+  Your answer: A function is an expression of code telling the computer what to do. However a function needs to be invoked by a console.log. A method tells the computer what to do with the predetermined parameters and does not need an invoking command such as console.log. If there is one line of code using a method, the computer will automatically return an output. If there is more than one expected output or new variables need to be established then a "return" must be declared. 
 
-  Researched answer:
+  Researched answer: A function is a set of instructions telling the computer how to do a specific task. A function has to be told when to execute (using console.log) at a certain time and how often. The logic of the function should be encapsulated and not interact with the rest of the program. A method is when a function belongs to an object.  An object is a collection of data/ functionality consisting of key value pairs. The method can create a variable to access the value within the object. For example, if there was an object of a dog (const dog) that consisted of the key value pairs of breed: "Shiba Inu" and age: 9, creating a function named dogData using string interpolation would return the breed and age of the dog when the method dogData was called upon. An example would be 
+  const dog = {
+    breed: "Shiba Inu",
+    age: 9,
+    dogData: function() {
+      return `${this.breed} is ${this.age}'s old`
+    }
+  }
+  console.log(dog.dogData())
+   
 
 
 
 5. What is object destructuring?
 
-  Your answer:
+  Your answer: Object destructuring is putting the nested classes into a method. When the object is called on to run, instead of invoking multiple nested items, the destructured object is ran. For example, if I was making a smoothie, which is the object, with multiple classes (categories) of ingredients, fruits, liquid base, boost, and extras, there are mulitple items in each of those classes (categories). Putting the empty class name in the method instead of specifying the individual value will allow the object to be specified and have an esthetically pleasing code.  
 
-  Researched answer:
+  Researched answer: Object destructuring is taking the property values of a variable and turning them into individual seperate variables that can be recalled. For example, if a variable such as a dog had their name and age as their key value pairs, destructuring each value would result in dog.name and dog.age. With the dog object destructured, recalling each value specific to the situation is simpler and efficient. 
```

Comment:
With an exception of the missed research answer on number 6, this entire section is EXACTLY what I am looking for!  Outstanding work on this section!  You answers were really thorough and well thought out.  Please make sure to volunteer one of these answers when we go through this in class 👏